### PR TITLE
feat: import repo page

### PR DESCRIPTION
**Pull request by shaurya-harness**
feat: import repo page

Screenshot 2024-12-13 at 3 26 04 PM
Screenshot 2024-12-16 at 1 43 52 PM
Screenshot 2024-12-16 at 2 47 42 PM

### CODE REVIEW

File: packages/ui/src/views/repo/repo-import/index.tsx
Comment (knagurski) [inline packages/ui/src/views/repo/repo-import/index.tsx:1]: Can we avoid using `index` files for components/types/logic etc, only use them as indexes to get at things more easily. Otherwise it's super annoying to find what you need quickly without having to traverse the directories. Ideally a file should export a single component and the name of the file should match that.

File: packages/ui/src/views/repo/repo-list/repo-list.tsx [hunk 1]
```diff
@@ -100,25 +100,27 @@ export function RepoList({
...
-                }
-                description={<Stats stars={repo.stars} pulls={repo.pulls} />}
-                right
-                label
-                secondary
-              />
+              {!repo.importing ? (
+                <StackedList.Field
+                  title={
+                    <>
```
Comment (knagurski) [inline packages/ui/src/views/repo/repo-list/repo-list.tsx:114]: Doesn't look like this fragment is needed

File: packages/ui/src/views/repo/repo-list/repo-list.tsx [hunk 2]
```diff
@@ -100,25 +100,27 @@ export function RepoList({
...
+                      Updated <em>{repo.timestamp}</em>
```
Comment (knagurski) [inline packages/ui/src/views/repo/repo-list/repo-list.tsx:115]: Can we get this into the strings?

Comment (shaurya-harness) [reply packages/ui/src/views/repo/repo-list/repo-list.tsx:115]: there's quite a bit of strings in this component, I will move them all

File: packages/ui/src/views/repo/repo-import/index.tsx
Comment (shaurya-harness) [reply packages/ui/src/views/repo/repo-import/index.tsx:1]: updated

File: packages/ui/src/views/repo/repo-list/repo-list-page.tsx [hunk 3]
```diff
@@ -14,7 +15,16 @@ import { sortRepositories } from '../utils/sorting/repos'
 import { RepoList } from './repo-list'
 import { RepoListProps } from './types'
 
-const LinkComponent = ({ to, children }: { to: string; children: ReactNode }) => <Link to={to}>{children}</Link>
+const LinkComponent = ({ to, children, disabled }: { to: string; children: ReactNode; disabled?: boolean }) => (
```
Comment (abhinavrastogi-harness) [inline packages/ui/src/views/repo/repo-list/repo-list-page.tsx:18]: why is this needed?
if needed, LinkComponent should be defined in the app layer and passed to the view.
View should not be defining a LinkComponent

Comment (shaurya-harness) [reply packages/ui/src/views/repo/repo-list/repo-list-page.tsx:18]: yes not required, probably transferred from old views of playground, removing this

File: packages/ui/src/views/repo/repo-list/repo-list.tsx (hunk 1 above)
Comment (shaurya-harness) [reply packages/ui/src/views/repo/repo-list/repo-list.tsx]: fixed

**Review by vardanbansal-harness — APPROVED** [review]
